File: frontend/src/types/insurance.ts
export enum InsuranceTypeEnum {
  PROPERTY = 'PROPERTY',
  LIABILITY = 'LIABILITY',
  VEHICLE = 'VEHICLE'
}

export enum InsuranceStatusEnum {
  ACTIVE = 'ACTIVE',
  EXPIRED = 'EXPIRED',
  CANCELLED = 'CANCELLED'
}

export enum ClaimStatusEnum {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED'
}

export interface Insurance {
  id?: number;
  policyNo: string;
  insuranceName: string;
  insuranceType: InsuranceTypeEnum;
  assetIds: string;
  assetIdList?: number[];
  insurer: string;
  premium: number;
  coverage?: number;
  deductible?: number;
  startDate: string;
  endDate: string;
  status: InsuranceStatusEnum;
  remark?: string;
  tenantId?: string;
  createBy?: number;
  createTime?: string;
  updateTime?: string;
  deleted?: number;
}

export interface InsuranceClaim {
  id?: number;
  claimNo: string;
  insuranceId: number;
  claimDate: string;
  incidentDescription?: string;
  claimAmount: number;
  settledAmount?: number;
  status: ClaimStatusEnum;
  settleDate?: string;
  remark?: string;
  policyNo?: string;
  insuranceName?: string;
  tenantId?: string;
  createBy?: number;
  createTime?: string;
  updateTime?: string;
  deleted?: number;
}

export interface InsuranceQueryParams {
  keyword?: string;
  insuranceType?: InsuranceTypeEnum;
  status?: InsuranceStatusEnum;
  startDate?: string;
  endDate?: string;
  pageNum?: number;
  pageSize?: number;
}

File: backend/src/main/java/com/ams/entity/Insurance.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("insurance")
public class Insurance {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String policyNo;
    private String insuranceName;
    private String insuranceType;
    private String assetIds;

    @TableField(exist = false)
    private List<Long> assetIdList;

    private String insurer;
    private BigDecimal premium;
    private BigDecimal coverage;
    private BigDecimal deductible;
    private LocalDate startDate;
    private LocalDate endDate;
    private String status;
    private String remark;

    private String tenantId;
    private Long createBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}